### ISSUE TITLE
Provisioning of custom configuration files

### ISSUE THREAD

**Issue by Ofinka** [elastic/cloud-on-k8s#1571]
## Bug Report
**What did you do?**
Following https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-bundles-plugins.html I tried to provision synonyms config files to elasticsearch container.

**What did you expect to see?**
The volume specified in podTemplate to be mounted along with all the configuration files.

**What did you see instead? Under which circumstances?**
My volume containing configuration files is not mounted at all.

**Environment**

* Version information:
ECK: 0.9.0
ES: 6.8.1

* Kubernetes information:
	* On premise 
```
$ kubectl version
```
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.4", GitCommit:"a87e9a978f65a8303aa9467537aa59c18122cbf9", GitTreeState:"clean", BuildDate:"2019-07-08T08:51:16Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.4", GitCommit:"a87e9a978f65a8303aa9467537aa59c18122cbf9", GitTreeState:"clean", BuildDate:"2019-07-08T08:43:10Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```
* Resource definition:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: ch-elasticsearch-config
  namespace: {{ .Values.global.namespace }}
data:
  synonyms-en.txt: |
    welle => shaft,
    motor => engine

  synonyms-de.txt: |
    shaft => welle,
    transmission => umformerelement, getrieb  
---

apiVersion: elasticsearch.k8s.elastic.co/v1alpha1
kind: Elasticsearch
metadata:
  name: elasticsearch
  namespace: {{ .Values.global.namespace }}
spec:
  podTemplate:
    spec:
      containers:
      - name: elasticsearch
        volumeMounts:
          - name: ch-elasticsearch-volume0
            mountPath: /usr/share/elasticsearch/config/dictionaries
      volumes:
        - name: ch-elasticsearch-volume0
          configMap:
            name: ch-elasticsearch-config
  version: {{ .Values.global.version }}
  http:
    service:
      spec:
        type: NodePort
    tls:
      certificate:
        secretName: ch-elasticsearch-tls
  nodes:
    - nodeCount: 1
      config:
        node.master: true
        node.data: true
        node.ingest: true
      volumeClaimTemplates:
        - metadata:
            name: elasticsearch-data
          spec:
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: {{ .Values.global.storageSize }}
            storageClassName: {{ .Values.global.storageClass }}
```

* Logs:
```
$ kubectl describe pod elasticsearch-es-p6jvmws672 -n operator-elk
Name:               elasticsearch-es-p6jvmws672
Namespace:          operator-elk
Priority:           0
PriorityClassName:  <none>
Node:               l1/10.10.31.10
Start Time:         Wed, 14 Aug 2019 10:41:16 +0200
Labels:             common.k8s.elastic.co/template-hash=3153862167
                    common.k8s.elastic.co/type=elasticsearch
                    elasticsearch.k8s.elastic.co/cluster-name=elasticsearch
                    elasticsearch.k8s.elastic.co/node-data=true
                    elasticsearch.k8s.elastic.co/node-ingest=true
                    elasticsearch.k8s.elastic.co/node-master=true
                    elasticsearch.k8s.elastic.co/node-ml=true
                    elasticsearch.k8s.elastic.co/version=6.8.1
Annotations:        update.k8s.elastic.co/timestamp: 2019-08-14T08:41:23.284675793Z
Status:             Running
IP:                 10.32.5.163
Controlled By:      Elasticsearch/elasticsearch
Init Containers:
  elastic-internal-init-os-settings:
    Container ID:  docker://eec2dd5f1b9b4a9314f846fb15f5bd9f129810a221a989119df53ea8fbef759d
    Image:         docker.elastic.co/elasticsearch/elasticsearch:6.8.1
    Image ID:      docker-pullable://docker.elastic.co/elasticsearch/elasticsearch@sha256:974aba875f1ec333f0c2ac7dee1e74d70c5813b039b3a225eb2bf287d4aa5911
    Port:          <none>
    Host Port:     <none>
    Command:
      sysctl
      -w
      vm.max_map_count=262144
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Wed, 14 Aug 2019 10:41:20 +0200
      Finished:     Wed, 14 Aug 2019 10:41:20 +0200
    Ready:          True
    Restart Count:  0
    Environment:
      POD_NAME:  elasticsearch-es-p6jvmws672 (v1:metadata.name)
      POD_IP:     (v1:status.podIP)
    Mounts:
      /mnt/elastic-internal/elasticsearch-config from elastic-internal-elasticsearch-config (ro)
      /mnt/elastic-internal/probe-user from elastic-internal-probe-user (ro)
      /mnt/elastic-internal/scripts from elastic-internal-scripts (ro)
      /mnt/elastic-internal/unicast-hosts from elastic-internal-unicast-hosts (ro)
      /mnt/elastic-internal/xpack-file-realm from elastic-internal-xpack-file-realm (ro)
      /usr/share/elasticsearch/bin from elastic-internal-elasticsearch-bin-local (rw)
      /usr/share/elasticsearch/config from elastic-internal-elasticsearch-config-local (rw)
      /usr/share/elasticsearch/config/http-certs from elastic-internal-http-certificates (ro)
      /usr/share/elasticsearch/config/transport-certs from elastic-internal-transport-certificates (ro)
      /usr/share/elasticsearch/data from elasticsearch-data (rw)
      /usr/share/elasticsearch/logs from elasticsearch-logs (rw)
      /usr/share/elasticsearch/plugins from elastic-internal-elasticsearch-plugins-local (rw)
  elastic-internal-init-filesystem:
    Container ID:  docker://05fd6fcfd5fc17cd8a8f227c37c00b9385adea52c898542de65c6dd40fa03476
    Image:         docker.elastic.co/elasticsearch/elasticsearch:6.8.1
    Image ID:      docker-pullable://docker.elastic.co/elasticsearch/elasticsearch@sha256:974aba875f1ec333f0c2ac7dee1e74d70c5813b039b3a225eb2bf287d4aa5911
    Port:          <none>
    Host Port:     <none>
    Command:
      bash
      -c
      /mnt/elastic-internal/scripts/prepare-fs.sh
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Wed, 14 Aug 2019 10:41:22 +0200
      Finished:     Wed, 14 Aug 2019 10:41:24 +0200
    Ready:          True
    Restart Count:  0
    Environment:
      POD_NAME:  elasticsearch-es-p6jvmws672 (v1:metadata.name)
      POD_IP:     (v1:status.podIP)
    Mounts:
      /mnt/elastic-internal/elasticsearch-bin-local from elastic-internal-elasticsearch-bin-local (rw)
      /mnt/elastic-internal/elasticsearch-config from elastic-internal-elasticsearch-config (ro)
      /mnt/elastic-internal/elasticsearch-config-local from elastic-internal-elasticsearch-config-local (rw)
      /mnt/elastic-internal/elasticsearch-plugins-local from elastic-internal-elasticsearch-plugins-local (rw)
      /mnt/elastic-internal/probe-user from elastic-internal-probe-user (ro)
      /mnt/elastic-internal/scripts from elastic-internal-scripts (ro)
      /mnt/elastic-internal/transport-certificates from elastic-internal-transport-certificates (ro)
      /mnt/elastic-internal/unicast-hosts from elastic-internal-unicast-hosts (ro)
      /mnt/elastic-internal/xpack-file-realm from elastic-internal-xpack-file-realm (ro)
      /usr/share/elasticsearch/config/http-certs from elastic-internal-http-certificates (ro)
      /usr/share/elasticsearch/data from elasticsearch-data (rw)
      /usr/share/elasticsearch/logs from elasticsearch-logs (rw)
Containers:
  elasticsearch:
    Container ID:   docker://535a9fe7a6fda4747ad91a29db67eff4928b4e63d55c567fbaa125830aac5eb6
    Image:          docker.elastic.co/elasticsearch/elasticsearch:6.8.1
    Image ID:       docker-pullable://docker.elastic.co/elasticsearch/elasticsearch@sha256:974aba875f1ec333f0c2ac7dee1e74d70c5813b039b3a225eb2bf287d4aa5911
    Ports:          9200/TCP, 9300/TCP
    Host Ports:     0/TCP, 0/TCP
    State:          Running
      Started:      Wed, 14 Aug 2019 10:41:26 +0200
    Ready:          True
    Restart Count:  0
    Requests:
      memory:   2Gi
    Readiness:  exec [bash -c /mnt/elastic-internal/scripts/readiness-probe-script.sh] delay=10s timeout=5s period=5s #success=1 #failure=3
    Environment:
      POD_IP:                     (v1:status.podIP)
      POD_NAME:                  elasticsearch-es-p6jvmws672 (v1:metadata.name)
      PROBE_PASSWORD_FILE:       /mnt/elastic-internal/probe-user/elastic-internal-probe
      PROBE_USERNAME:            elastic-internal-probe
      READINESS_PROBE_PROTOCOL:  https
    Mounts:
      /mnt/elastic-internal/elasticsearch-config from elastic-internal-elasticsearch-config (ro)
      /mnt/elastic-internal/probe-user from elastic-internal-probe-user (ro)
      /mnt/elastic-internal/scripts from elastic-internal-scripts (ro)
      /mnt/elastic-internal/unicast-hosts from elastic-internal-unicast-hosts (ro)
      /mnt/elastic-internal/xpack-file-realm from elastic-internal-xpack-file-realm (ro)
      /usr/share/elasticsearch/bin from elastic-internal-elasticsearch-bin-local (rw)
      /usr/share/elasticsearch/config from elastic-internal-elasticsearch-config-local (rw)
      /usr/share/elasticsearch/config/http-certs from elastic-internal-http-certificates (ro)
      /usr/share/elasticsearch/config/transport-certs from elastic-internal-transport-certificates (ro)
      /usr/share/elasticsearch/data from elasticsearch-data (rw)
      /usr/share/elasticsearch/logs from elasticsearch-logs (rw)
      /usr/share/elasticsearch/plugins from elastic-internal-elasticsearch-plugins-local (rw)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  elastic-internal-elasticsearch-bin-local:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  elastic-internal-elasticsearch-config:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  elasticsearch-es-p6jvmws672-config
    Optional:    false
  elastic-internal-elasticsearch-config-local:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  elastic-internal-elasticsearch-plugins-local:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  elastic-internal-http-certificates:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  elasticsearch-es-http-certs-internal
    Optional:    false
  elastic-internal-probe-user:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  elasticsearch-es-internal-users
    Optional:    false
  elastic-internal-scripts:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      elasticsearch-es-scripts
    Optional:  false
  elastic-internal-transport-certificates:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  elasticsearch-es-p6jvmws672-certs
    Optional:    false
  elastic-internal-unicast-hosts:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      elasticsearch-es-unicast-hosts
    Optional:  false
  elastic-internal-xpack-file-realm:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  elasticsearch-es-xpack-file-realm
    Optional:    false
  elasticsearch-data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  elasticsearch-es-p6jvmws672-elasticsearch-data
    ReadOnly:   false
  elasticsearch-logs:
    Type:        EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:   <unset>
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason            Age                From                   Message
  ----     ------            ----               ----                   -------
  Warning  FailedScheduling  29m (x2 over 29m)  default-scheduler      pod has unbound immediate PersistentVolumeClaims
  Normal   Scheduled         29m                default-scheduler      Successfully assigned operator-elk/elasticsearch-es-p6jvmws672 to l1
  Normal   Pulled            29m                kubelet, l1  Container image "docker.elastic.co/elasticsearch/elasticsearch:6.8.1" already present on machine
  Normal   Created           29m                kubelet, l1  Created container elastic-internal-init-os-settings
  Normal   Started           29m                kubelet, l1  Started container elastic-internal-init-os-settings
  Normal   Pulled            29m                kubelet, l1  Container image "docker.elastic.co/elasticsearch/elasticsearch:6.8.1" already present on machine
  Normal   Created           29m                kubelet, l1  Created container elastic-internal-init-filesystem
  Normal   Started           29m                kubelet, l1  Started container elastic-internal-init-filesystem
  Normal   Pulled            29m                kubelet, l1  Container image "docker.elastic.co/elasticsearch/elasticsearch:6.8.1" already present on machine
  Normal   Created           29m                kubelet, l1  Created container elasticsearch
  Normal   Started           29m                kubelet, l1  Started container elasticsearch
  Warning  Unhealthy         29m (x2 over 29m)  kubelet, l1  Readiness probe failed:
```

**Comment by pebrc** [comment]
This is a very unfortunate omission in the docs, sorry about that 😞 
The pod template is node spec specific so it should be:
```yaml
spec:
  nodes: 
  - name: my-node 
    podTemplate:
      spec:
        initContainers:
        - name: install-plugins
          command:
          - sh
          - -c
          - |
            bin/elasticsearch-plugin install --batch repository-azure
```
We will fix the docs.

**Comment by Ofinka** [comment]
Works fine with your example. Thanks for clarification :)

**Comment by pebrc** [comment]
Fixed in #1574
